### ISSUE TITLE
[BUG] Power on 3 harvester nodes in sequence may cause VIP connectivity lost

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#1681]
**Describe the bug**

Power on 3 harvester nodes in sequence have possibility failed to access VIP
Need to wait for long time (in this case, wait 20~30 min) to be able to recover connection 

```
hpd8s7:/home/rancher # ping 10.84.44.181
PING 10.84.44.181 (10.84.44.181) 56(84) bytes of data.
From 10.84.44.114 icmp_seq=1 Destination Host Unreachable
From 10.84.44.114 icmp_seq=2 Destination Host Unreachable
From 10.84.44.114 icmp_seq=3 Destination Host Unreachable
```

**To Reproduce**
1. Prepare 3 nodes harvester
2. Enable virtual network with `harvester-mgmt`
3. create vlan network  `vlan 806` with id `806`
4. create several vm
5. Provision a RKE2 cluster 
6. Press 'Momentary Press` button to shutdown node 3
7. Press 'Momentary Press` button to shutdown node 2
8. Press 'Momentary Press` button to shutdown node 1
9. Wait for 20 minutes
10. Press 'Momentary Press` button to power on node 1
11. Press 'Momentary Press` button to power on node 2
12. Press 'Momentary Press` button to power on node 3
13. Wait for startup complete
14. Check node and VIP accessibility

**Expected behavior**

Should be able to access VIP 

**Support bundle**

[supportbundle_29976b55-cbd5-4e7d-a7a2-9c883c9b375c_2021-12-15T09-47-35Z.zip](https://github.com/harvester/harvester/files/7718167/supportbundle_29976b55-cbd5-4e7d-a7a2-9c883c9b375c_2021-12-15T09-47-35Z.zip)

**Environment:**
 - Harvester ISO version: master-5d6f3db0-head
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 3 nodes harvester on provo bare machines

**Server information:**
 - VIP: 10.84.44.181
 - node 1: 10.84.44.114
 - node 2: 10.84.44.115
 - node 3: 10.84.44.116

**Additional context**
Not always happen. There are two ways to solve if encounter this issue
1. Wait for a while, the kube-vip pod may recover the VIP connectivity
2. Kill kube-vip pod

**Comment by yaocw2020** [comment]
Log from kube-vip
```
2021-12-15T07:49:31.797455726Z time="2021-12-15T07:49:31Z" level=info msg="New VIP [10.84.44.181] for [ingress-expose/1b1c95c6-2a0d-4b0e-a4d2-0fd5f943cec5] "
2021-12-15T07:49:31.797583912Z time="2021-12-15T07:49:31Z" level=error msg="Failed to add Service [ingress-expose] / [1b1c95c6-2a0d-4b0e-a4d2-0fd5f943cec5]"
2021-12-15T07:49:31.797626757Z time="2021-12-15T07:49:31Z" level=error msg="could not get link for interface 'harvester-br0': Link not found"
```
Log from harvester-network-controller
```
I1215 07:50:07.776925       1 controller.go:69] nad configuration vlan806 has been changed: {"cniVersion":"0.3.1","name":"vlan806","type":"bridge","bridge":"harvester-br0","promiscMode":true,"vlan":806,"ipam":{}}
I1215 07:50:07.777079       1 controller.go:85] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
time="2021-12-15T07:50:07Z" level=info msg="Starting network.harvesterhci.io/v1beta1, Kind=NodeNetwork controller"
I1215 07:50:07.777186       1 controller.go:95] node network configuration harvester-sk7tr-vlan has been changed, spec: {Description: NodeName:harvester-sk7tr Type:vlan NetworkInterface:harvester-mgmt}
I1215 07:50:07.777348       1 controller.go:172] ignore link not found error, details: fetch bridge harvester-br0 failed, error: could not lookup link harvester-br0, error: Link not found
I1215 07:50:07.777548       1 controller.go:280] add nad:vlan806 with vid:806 to the list
I1215 07:50:07.777565       1 vlan.go:88] start to setup VLAN network
I1215 07:50:09.637456       1 bridge.go:69] add IPv4 address 10.84.44.116/23 harvester-br0
I1215 07:50:09.637579       1 link.go:267] ignore existing route: {Ifindex: 20 Dst: 10.84.44.0/23 Src: 10.84.44.116 Gw: <nil> Flags: [] Table: 254}
I1215 07:50:09.637695       1 link.go:265] append route: {Ifindex: 20 Dst: <nil> Src: <nil> Gw: 10.84.45.254 Flags: [] Table: 254}
I1215 07:50:15.935969       1 link.go:279] delete route {Ifindex: 5 Dst: 10.84.44.0/23 Src: 10.84.44.116 Gw: <nil> Flags: [] Table: 254}
I1215 07:50:15.936123       1 link.go:279] delete route {Ifindex: 5 Dst: <nil> Src: <nil> Gw: 10.84.45.254 Flags: [] Table: 254}
I1215 07:50:15.936184       1 vlan.go:123] setup VLAN network successfully
```

From the logs, we can see that the kube-vip created the macvlan interface on the base of `harvester-br0` which was not created because the  harvester network controller has not set up the VLAN network.

**Comment by yaocw2020** [comment]
Work around: Delete the kube-vip pod to let it retry to get the VIP.

**Comment by TachunLin** [comment]
In case of user encounter the situation that they can't access harvester dashboard by VIP but they can still access (ssh) to each node ip. They can use the delete all kube-vip pod to make the VIP come back. 

> But please be notes, it based on you still can run `kubectl get pods` command 

```
harvester-node02-1216:/home/rancher # kubectl get pods -A | grep kube-vip
harvester-system kube-vip-cloud-provider-0 1/1 Running 4 88m
harvester-system kube-vip-f97km 1/1 Running 0 2m43s
harvester-system kube-vip-j79zd 1/1 Running 0 3m3s
harvester-system kube-vip-wbvhc 1/1 Running 0 3m55s

harvester-node02-1216:/home/rancher # kubectl delete pods -n harvester-system kube-vip-f97km
pod "kube-vip-f97km" deleted
harvester-node02-1216:/home/rancher # kubectl get pods -A | grep kube-vip
harvester-system kube-vip-cloud-provider-0 1/1 Running 4 92m
harvester-system kube-vip-j79zd 1/1 Running 0 7m11s
harvester-system kube-vip-jnsp9 1/1 Running 0 10s
harvester-system kube-vip-wbvhc 1/1 Running 0 8m3s
```

**Comment by yaocw2020** [comment]
https://github.com/kube-vip/kube-vip/pull/329

**Comment by TachunLin** [comment]
Verified **fixed** on master-1be9cb5e-head. Close this issue.

### Result
Can access 3 nodes harvester by VIP after shutdown and power on harvester node machine in sequence

```
rancher@hpd8s7:~> sudo -i kubectl get pods -A | grep kube-vip
harvester-system            kube-vip-8lzhd                                          1/1     Running            5          6h51m
harvester-system            kube-vip-cloud-provider-0                               1/1     Running            1          6h26m
harvester-system            kube-vip-lgbgb                                          1/1     Running            5          6h52m
harvester-system            kube-vip-txbp7                                          1/1     Running            4          6h51m
```

```
rancher@hpd8s7:~> ip a | grep 10.84
    inet 10.84.44.114/23 brd 10.84.45.255 scope global harvester-mgmt
    inet 10.84.44.124/32 scope global harvester-mgmt
    inet 10.84.44.114/23 brd 10.84.45.255 scope global harvester-br0
    inet 10.84.44.124/32 scope global harvester-br0
```

```
hpd8s13:~ # ping 10.84.44.124
PING 10.84.44.124 (10.84.44.124) 56(84) bytes of data.
64 bytes from 10.84.44.124: icmp_seq=1 ttl=64 time=0.233 ms
64 bytes from 10.84.44.124: icmp_seq=2 ttl=64 time=0.150 ms
64 bytes from 10.84.44.124: icmp_seq=3 ttl=64 time=0.139 ms
```


### Test Information
* Test Environment: 3 nodes harvester on provo bare machine
* Harvester version: master-1be9cb5e-head (1/19)


### Verify Steps

1. Prepare 3 nodes harvester
2. Enable virtual network with harvester-mgmt
3. create vlan network vlan 806 with id 806
4. create several vm
5. Provision a RKE2 cluster
6. Press 'Momentary Press` button to shutdown node 3
7. Press 'Momentary Press` button to shutdown node 2
8. Press 'Momentary Press` button to shutdown node 1
9. Wait for 20 minutes
10. Press 'Momentary Press` button to power on node 1
11. Press 'Momentary Press` button to power on node 2
12. Press 'Momentary Press` button to power on node 3
13. Wait for startup complete
14. Check node and VIP accessibility

### Additional Context
Workaround if encounter VIP connectivity issue:
1. ssh to controller node
2. Run `kubectl rollout restart daemonset kube-vip -n harvester-system`
